mlflow/server/job_api.py

```diff
@@ -90,3 +99,40 @@ def submit_job(payload: SubmitJobPayload) -> Job:
             status_code=e.get_http_status_code(),
             detail=e.message,
         )
+
+
+class SearchJobPayload(BaseModel):
+    function_fullname: str
+    params: dict[str, Any] | None = None
+    statuses: list[str] | None = None
+
+
+@job_api_router.post("/search", response_model=JobList)
+def search_job(payload: SearchJobPayload):
+    from mlflow.server.handlers import _get_job_store
+
+    try:
+        statuses = [JobStatus.from_str(status) for status in (payload.statuses or [])]
+        expected_params = payload.params
+
+        def filter_params(params: dict[str, Any]) -> bool:
```

Comment:
why not filter using sqlalchemy?

Reply:
The params is a dict with arbitrary keys (to support generic job params), and saved in database as serialized json string, so that can't using SQL to filter.

Reply:
Sorry, I didn't read this block carefully. Makes sense.